crates/cheatcodes/src/inspector.rs

```diff
@@ -590,23 +590,33 @@ impl<DB: DatabaseExt + Send> Inspector<DB> for Cheatcodes {
     }
 
     fn step_end(&mut self, interpreter: &mut Interpreter<'_>, data: &mut EVMData<'_, DB>) {
-        if self.use_zk_vm && interpreter.current_opcode() == opcode::BALANCE {
-            if interpreter.stack.is_empty() {
-                interpreter.instruction_result = InstructionResult::StackUnderflow;
-                return;
-            }
-            // Safety: Length is checked above.
-            let address = Address::from_word(B256::from(unsafe { interpreter.stack.pop_unsafe() }));
-            let balance = foundry_zksync_core::balance(address, data.db, &mut data.journaled_state);
-
-            // Skip the current BALANCE instruction since we've already handled it
-            match interpreter.stack.push(balance) {
-                Ok(_) => unsafe {
-                    interpreter.instruction_pointer = interpreter.instruction_pointer.add(1);
-                },
-                Err(e) => {
-                    interpreter.instruction_result = e;
+        if !self.use_zk_vm {
+            return;
+        }
```

Comment:
Not sure if I'm a fan of this early return - mostly because we should be scoping the zksync logic in the cheatcode inspector, and `step_end` may be used by foundry as well. 
I'm a bit concerned that this early return is a bit too "loose" and doesn't immediately capture the zksync specific logic we've added to the cheatcode inspector. I believe it's more intuitive to have the opcode override logic scoped within the if, than change the `step_end` hook's context.